### DISCUSSION TITLE
Memory consumption (docker)

### DISCUSSION THREAD

**Discussion by DuvelCorp** [gotson/komga#1150] [General]
Hi,

I've got a pretty decent army of docker containers to run my media center / seedbox  infra, and I notice that Komga is far ahead the pack regarding memory consumption, and that's not a good thing.

You can see this on this graph showing the memory usage of all my containers dispatched on 3 linux servers

![image](https://github.com/gotson/komga/assets/33682764/203a9f05-f4c1-4b18-8873-08da59bf6c5c)

That's the last 24 hours and during this time frame no users connected to Komga, and the library files haven't changed at all. In fact, the library files have not changed for the last 6 months.

These are the metrics in Komga, not sure if you consider this as a big library or not.
![image](https://github.com/gotson/komga/assets/33682764/5644ca00-227d-4f25-acfc-26ce8c5a887a)

The log file doesn't show anything special
![image](https://github.com/gotson/komga/assets/33682764/ceb31f5f-79f7-4aa0-88f4-ee44a988bde0)

We can see that it had some maintenance tasks running around midnight and then nothing during the whole night. The 07:24 activity is me now when I connected to get the metrics.
Nevertheless, it kept using 700Mb of RAM...
Is this normal behavior??? Since the log file doesn't show anything,  I have the impression that Komga is always retaining stuff in memory for absolutely nothing.

Can this be improved in the core or tuned on my side ??

Cheers

**Comment by gotson** [comment]
Check https://komga.org/faq/#the-memory-consumption-is-huge



> Nevertheless, it kept using 700Mb of RAM...

It's fairly normal for a Java application.

**Comment by protofolius** [comment]
If you are looking for a comic server that uses almost no memory and is blazing fast : codex .

Even though i would also like komga to use less memory  , as gotson mentions : komga is built on java . and java is known to eat lots of memory . :\
Maybe it would be nice to get komga into a sort of hibernate state where it goes to the absolute minimum after no more than 1 user has connected in a period of time . or a setting that offloads to disk and frees memory . something , options :)

**Reply by DuvelCorp** [reply]
@protofolius txs
Not sure that codex is able to handle correctly the "BD Franco-Belge" comics which is the only kind of stuff I have. 
Komga handles it perfectly
From their test site it looks like codex is more US comics based.
But I'll give it a try
